src/ansys/fluent/core/post_objects/post_helper.py

```diff
@@ -131,11 +134,21 @@ def get_vector_fields(self):
 
     def get_field_unit(self, field):
         """Returns the unit of the field."""
-        quantity = self._field_unit_quantity(field)
-        if quantity == "*null*":
-            return ""
-        scheme_eval_str = f"(units/get-pretty-wb-units-from-dimension (units/inquire-dimension '{quantity}))"
-        return " ".join(self._scheme_str_to_py_list(scheme_eval_str))
+        session = self.obj.get_root().session
+        if hasattr(session, "scheme_eval"):
+            if FluentVersion(session.scheme_eval.version) < FluentVersion.v252:
```

Comment:
Actually, it may be possible to encapsulate this inside FieldData service class.

class FieldData:
    def __init__(..., version, scheme_eval=None):
    ...
    def get_field_info(...):
    # implement either through scheme_eval or field_data service based on version

Reply:
@mkundu1 We are not using scheme eval to get field info. We are using scheme eval to get units only. We have encapsulated get_field_info in the FieldInfo class already.